src/Tribe/Template/Month.php

```diff
@@ -421,42 +426,105 @@ protected function set_events_in_month() {
+				$this->event_ids_by_day = array();
 
-			$beginning_of_day           = tribe_event_beginning_of_day( $date );
-			$beginning_of_day_timestamp = strtotime( $beginning_of_day );
+				// Let's loop over all of the events in the month and assign them to days
+				foreach ( $this->events_in_month as $event ) {
+					// if we're querying by category and the event doesn't have it, skip the event
+					if ( ! empty ( $this->queried_event_cats ) ) {
+						if ( ! has_term( $this->queried_event_cats, Tribe__Events__Main::TAXONOMY, $event ) ) {
+							continue;
+						}
+					}
 
-			$end_of_day           = tribe_event_end_of_day( $date );
-			$end_of_day_timestamp = strtotime( $end_of_day );
+					$event_start = strtotime( $event->EventStartDate );
+					$event_end   = strtotime( $event->EventEndDate );
 
-			$event_ids_on_date = array();
+					$start = date( 'Y-m-d', $event_start );
+					$end = date( 'Y-m-d', $event_end );
 
-			// loop through all the events in the month and find the ones on the requested date
-			foreach ( $this->events_in_month as $event ) {
+					$beginning_of_day           = tribe_event_beginning_of_day( $start );
+					$beginning_of_day_timestamp = strtotime( $beginning_of_day );
 
-				$event_start = strtotime( $event->EventStartDate );
-				$event_end   = strtotime( $event->EventEndDate );
+					$end_of_day           = tribe_event_end_of_day( $end );
+					$end_of_day_timestamp = strtotime( $end_of_day );
 
-				// check if the event happens on this day
-				if ( Tribe__Events__Date_Utils::range_coincides( $beginning_of_day_timestamp, $end_of_day_timestamp, $event_start, $event_end ) ) {
-					// check if the event has the term being viewed, if not, skip it
-					if ( ! empty ( $this->queried_event_cats ) ) {
-						if ( ! has_term( $this->queried_event_cats, Tribe__Events__Main::TAXONOMY, $event ) ) {
-							continue;
+					// if the start of the event is earlier than the beginning of the day, consider the event as starting on the day before
+					if ( $event_start < $beginning_of_day_timestamp ) {
+						$start = date( 'Y-m-d', strtotime( '-1 day', strtotime( $start ) ) );
+					}
+
+					// if the end of the event is later than the end of the day, consider the event as ending on the next day
+					if ( $event_end > $end_of_day_timestamp ) {
+						$end = date( 'Y-m-d', strtotime( '+1 day', strtotime( $end ) ) );
+					}
+
+					// determine if there's a difference in days between start and end
+					$diff = strtotime( $end ) - strtotime( $start );
+
+					if ( $diff > 0 ) {
+						// There IS a difference. How many days?
+						$diff_in_days = $diff / DAY_IN_SECONDS;
+
+						// add the event to each day until the event end
+						$new_start = $start;
+						for ( $i = 0; $i <= $diff_in_days; $i++ ) {
+							if ( ! isset( $this->event_ids_by_day[ $new_start ] ) ) {
+								$this->event_ids_by_day[ $new_start ] = array();
+							}
+
+							$this->event_ids_by_day[ $new_start ][] = $event->ID;
```

Comment:
Hey Matt: got a slight concern about this re EOD cutoffs.

Looking back to line 471, `$start` and `$end` are just dates (rather than complete datetimes) - so, if execution has entered the block beginning line 473, `$diff_in_days` will be at least `1` and this `for` loop is going to run for at least 2 iterations.

Given an event with a late finish (ie, the next day but _within_ the cutoff period) the ID will be added incorrectly to that final day - throwing the count of events on that day (ie, we might see *"View all X events"* where X is one unit higher than it should be).